vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs

```diff
@@ -726,4 +728,216 @@ type internal FSharpNavigation
+            let entityPath = parts[..(parts.Length - 2)] |> List.ofArray
+            let memberOrVal = parts[parts.Length - 1]
+            DocCommentId.Field { EntityPath = entityPath; MemberOrValName = memberOrVal; GenericParameters = 0 }
+        | _ -> DocCommentId.None
+    
+    let tryFindFieldByName (name: string) (e: FSharpEntity) =
+        e.FSharpFields
+        |> Seq.filter (
+            fun x ->
+                x.DisplayName = name
+                && not x.IsCompilerGenerated)
+        |> Seq.map (fun e -> e.DeclarationLocation)
+
+    let tryFindValByNameAndType (name: string) (symbolMemberType: SymbolMemberType) (genericParametersCount: int) (e: FSharpEntity) =
+
+        let entities = e.TryGetMembersFunctionsAndValues()
+        
+        let defaultFilter (e: FSharpMemberOrFunctionOrValue) =
+            (e.DisplayName = name || e.CompiledName = name) && e.GenericParameters.Count = genericParametersCount
+
+        let  getLocation (e: FSharpMemberOrFunctionOrValue) = e.DeclarationLocation
+
+        let filteredEntities: range seq = 
+            match symbolMemberType with
+            | SymbolMemberType.Constructor when e.IsFSharpRecord ->
+                Seq.singleton e.DeclarationLocation
+            | SymbolMemberType.Other
+            | SymbolMemberType.Method ->
+                entities
+                |> Seq.filter defaultFilter
+                |> Seq.map getLocation
+            | SymbolMemberType.Constructor ->
+                entities
+                |> Seq.filter (fun x -> defaultFilter x && x.IsConstructor)
+                |> Seq.map getLocation
+            // When navigating to property for the record, it will be in members bag for custom ones, but will be in the fields in fields.
+            | SymbolMemberType.Property when e.IsFSharpRecord ->
+                let properties = 
+                    entities
+                    |> Seq.filter (fun x -> defaultFilter x && x.IsProperty)
+                    |> Seq.map getLocation
+                let fields = tryFindFieldByName name e
+                Seq.append properties fields
+            | SymbolMemberType.Event // Events are just properties
+            | SymbolMemberType.Property ->
+                entities
+                |> Seq.filter (fun x -> defaultFilter x && x.IsProperty)
+                |> Seq.map getLocation
+        filteredEntities
+
+    interface IFSharpCrossLanguageSymbolNavigationService with
+        member _.TryGetNavigableLocationAsync(assemblyName: string, documentationCommentId: string, cancellationToken: CancellationToken) : Task<IFSharpNavigableLocation> =
+            let path = docCommentIdToPath documentationCommentId
+            backgroundTask {
+                let projects = workspace.CurrentSolution.Projects |> Seq.filter (fun p -> p.IsFSharp && p.AssemblyName = assemblyName)
+                
+                let mutable locations = Seq.empty
+
+                for project in projects do
```

Comment:
We could probably do this in parallel.

Reply:
This is just a fool-proof, we expect it always be only a single project, parallel then will give you a bunch of overhead

Reply:
Right, didn't notice where it's coming from. Is it even theoretically possible there will be multiple projects with the same assembly name? Because maybe you could replace `filter` with `tryFind` and it would be a bit simpler.

Reply:
> Is it even theoretically possible there will be multiple projects with the same assembly name? 

Very unlikely, I have never seen it before.

> Because maybe you could replace `filter` with `tryFind` and it would be a bit simpler.

I'll see how it looks with option

Reply:
With option it could look something like this (After also adding `DocCommentId.EntityPath`):
```fsharp
let project = workspace.CurrentSolution.Projects |> Seq.tryFind (fun p -> p.IsFSharp && p.AssemblyName = assemblyName)
let locations = 
    project
    |> Option.map (fun project -> 
        let! checker, _, _, options = project.GetFSharpCompilationOptionsAsync(cancellationToken)
        let! result = checker.ParseAndCheckProject(options)
        let entity = path.EntityPath |> result.AssemblySignature.FindEntityByPath
        match path with
        | DocCommentId.Member ({ MemberOrValName = memberOrVal; GenericParameters = genericParametersCount }, memberType) ->
            entity |> tryFindVal memberOrVal documentationCommentId memberType genericParametersCount
        | DocCommentId.Field { MemberOrValName = memberOrVal } ->
            entity |> tryFindFieldByName memberOrVal
        | DocCommentId.Type _ -> 
            Seq.singleton entity.DeclarationLocation
        | DocCommentId.None -> 
            Seq.empty    
        |> Seq.map (fun m -> (m, project)))
    |> Option.defaultValue Seq.empty
```

A bit easier to see what's going on with arrows only going in one direction :)